runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ReadTranslator.java

```diff
@@ -54,9 +54,9 @@ public void translate(
     final PCollection<T> output = ctx.getOutput(transform);
     final Coder<WindowedValue<T>> coder = SamzaCoders.of(output);
     final Source<?> source =
-        transform instanceof Read.Unbounded
-            ? ((Read.Unbounded) transform).getSource()
-            : ((Read.Bounded) transform).getSource();
+        transform instanceof SplittableParDo.PrimitiveBoundedRead
```

Comment:
I was trying to figure out where the failure came from and I would assume here but I don't understand why.

Reply:
This is some pre-existing bug in the Samza runner that I'm exposing as all the other validates runner tests pass that used to pass and now they are using SDF based Reads everywhere (which covers Create as well).

Reply:
When debugging locally I can see that the PAssert executes once and gets the correct input and then it executes another time only containing `"finished"` (coming from the finishBundle on the TestFn) which implies that the PAssert trigger is occurring multiple times causing the verifier to fail on a subsequent bundle.